File: src/python/validators/email_validator.py
import re


class EmailValidator:

    pattern = r"^[a-z0-9][a-z0-9.]*@[a-z0-9]*\.?[a-z0-9.]*[a-z0-9]$"

    def is_valid(self, email):
        """Returns True for valid email and False for invalid email

        Args:
            email (str): Email string

        Returns:
            bool: Email validity
        """

        if re.match(EmailValidator.pattern, email):
            return True
        return False

File: src/ruby/app/validators/email_validator.rb
class EmailValidator
  def initialize(email)
    @email = email
  end

  def is_valid?
    has_only_allowed_characters and
    has_only_one_at_symbol and
    does_not_start_or_end_with_dot and
    has_text_after_at_symbol
  end

  def has_only_allowed_characters
    return false unless @email =~ /^[a-z0-9\.@]+$/
    true
  end

  def has_only_one_at_symbol
    return false unless @email.count('@') == 1
    true
  end

  def does_not_start_or_end_with_dot
    return true unless @email.start_with?('.') or @email.end_with?('.')
    false
  end

  def has_text_after_at_symbol
    return false unless @email =~ /@[a-z0-9\.]+$/
    true
  end
end
